### PR TITLE
fixing mysql query and database return values

### PR DESCRIPTION
**Pull request by nevins-b**
address #37 and some things in #23

### CODE REVIEW

File: phplib/Job.php [hunk 1]
```diff
@@ -135,14 +135,14 @@ public static function fail($date) {
             UPDATE `%s` SET `state` = ?, `update_date` = ?
             WHERE `site_id` = ? AND `archived` = 0 AND `state` IN %s AND `update_date` < ? AND `tries` > ?',
         $MODEL::$TABLE, DB::inPlaceholder(2));
-        DB::query($sql, [$MODEL::ST_CANC, $date, SiteFinder::getCurrentId(), $MODEL::ST_RUN, $MODEL::ST_FAIL, $threshold, $MODEL::MAX_TRIES]);
+        DB::query($sql, [$MODEL::ST_CANC, $date, SiteFinder::getCurrentId(), $MODEL::ST_RUN, $MODEL::ST_FAIL, $threshold, $MODEL::MAX_TRIES]), DB::CNT;
```
Comment (kiwiz) [inline phplib/Job.php]: That should be inside the parens.

File: phplib/REST/Dashboard.php [hunk 2]
```diff
@@ -39,16 +39,29 @@ public function GET(array $get) {
...
-            SELECT DATE(create_date, "unixepoch") as `date`, `action`, COUNT(*) as `count` FROM `%s` INNER JOIN (
-                SELECT DATE(create_date, "unixepoch"), `alert_id`, MAX(`create_date`) as `create_date`
-                FROM `%s` WHERE `site_id` = ? AND `archived` = 0 AND (
-                    (`action` = ? AND `a` = 1) OR
-                    (`action` = ? AND (`a` != 0 OR `b` != 0)) OR
-                    (`action` = ? AND `a` = 2)
-                ) AND `create_date` > ? AND DATE(create_date, "unixepoch") > DATE(?, "unixepoch") GROUP BY 1, 2
-            ) AS `tbl` USING(`alert_id`, `create_date`) GROUP BY 1, 2;
-        ', AlertLog::$TABLE, AlertLog::$TABLE);
+        if (DB::getType() == 'mysql') {
```
Comment (kiwiz) [inline phplib/REST/Dashboard.php]: Not a huge fan of this, but it works as an interim solution. #10 is intended to solve this.

Comment (nevins-b) [inline phplib/REST/Dashboard.php]: I agree, it's really ugly but works.

**Comment by nevins-b** [comment]
Yes, I was running into errors like
```
[+] Scheduler: 1475610183
[+] Maintenance
PHP Fatal error:  Uncaught PDOException: SQLSTATE[HY000]: General error in /var/www/411/phplib/DB.php:121
Stack trace:
#0 /var/www/411/phplib/DB.php(121): PDOStatement->fetchAll(2)
#1 /var/www/411/phplib/Element.php(147): FOO\DB::query('UPDATE `search_...', Array)
#2 /var/www/411/phplib/Scheduler.php(318): FOO\ElementFinder::reap(1475610183)
#3 /var/www/411/phplib/Scheduler.php(142): FOO\Scheduler->maintenance(1475610183)
#4 /var/www/411/bin/cron.php(45): FOO\Scheduler->processSite(Object(FOO\Site), 1475610183, false)
#5 {main}
  thrown in /var/www/411/phplib/DB.php on line 121
err [411_Scheduler] Scheduler error site:[1] ret:[65280]
```

Without the DB::CNT option

File: phplib/Job.php (hunk 1 above)
Comment (nevins-b) [inline phplib/Job.php]: oops, fixed.

**Comment by kiwiz** [comment]
I think I'll put the fix inside `DB.php` per this [StackOverflow question](https://stackoverflow.com/questions/16703896/pdo-using-fetchall-on-unknown-query). Can you update the pull request to contain just the `Dashboard.php` changes?

**Comment by kiwiz** [comment]
Merged, thanks!

**Pull request merged**